src/components/ifxRange/ifxRange.tsx

```diff
@@ -0,0 +1,85 @@
+import { Component, h, Prop, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-range',
+  styleUrl: 'ifxRange.scss',
+  shadow: true,
+})
+export class IfxRange {
+  @Prop() min: number = 0;
+  @Prop() max: number = 100;
+  @Prop({ mutable: true }) value: number = 0;
+  @Prop() disabled: boolean = false;
+  @Prop() showPercentage: boolean = false; // New property for toggleable percentage display
+  @Prop() leftIcon: string;
+  @Prop() rightIcon: string;
+  @Prop() leftText: string;
+  @Prop() rightText: string;
+  @Event() valueChanged: EventEmitter<number>;
+
+  private inputRef: HTMLInputElement;
+
+  handleInputChange(event: Event) {
+    const target = event.target as HTMLInputElement;
+    this.value = parseInt(target.value);
+    this.valueChanged.emit(this.value);
+    this.updateValuePercent();
+  }
+
+  updateValuePercent() {
+    if (this.inputRef) {
+      const percent = ((this.value - this.min) / (this.max - this.min)) * 100;
+      this.inputRef.style.setProperty('--value-percent', `${percent}%`);
+    }
+  }
+
+  componentDidLoad() {
+    this.updateValuePercent();
+  }
+
+
+  render() {
+    return (
+      <div class="ifx-range">
+        {this.leftText && (
+          <span class={`left-text${this.disabled ? ' disabled' : ''}`}>
+            {this.leftText}
+          </span>
+        )}
+        {this.leftIcon && (
+          <i class={`left-icon${this.disabled ? ' disabled' : ''}`}>
+            {this.leftIcon}
+          </i>
+        )}
+
+        <input
+          type="range"
+          min={this.min}
+          max={this.max}
+          disabled={this.disabled}
+          value={this.value}
+          ref={(el) => (this.inputRef = el as HTMLInputElement)}
+          onInput={(event) => this.handleInputChange(event)}
+        />
+        {this.rightIcon && (
+          <i class={`right-icon${this.disabled ? ' disabled' : ''}`}>
+            {this.rightIcon}
+          </i>
+        )}
+        {this.rightText && (
+          <span class={`right-text${this.disabled ? ' disabled' : ''}`}>
```

Comment:
Suggestion: `right-text${this.disabled ? ' disabled' : ''}` -> `right-text ${this.disabled ? 'disabled' : ''}`

Reply:
fixed

Reply:
wait ... i have to disagree ->
think about the result. With my solution its either "right-text" or "right-text disabled" with your solution it's "right-text " or "right-text disabled". There is a space to much. In the end it doesn't really matter but my solution is more consistent -> The space is only needed when disabled.